### ISSUE TITLE
Webhooks Fail Authentication Using Your Guide When Names Contain (&) Symbols

### ISSUE THREAD

**Issue by hecktarzuli** [storyblok/storyblok#783]
**The issue occurs while working with:** (check one with "x")   
- [ ] *app.storyblok.com (CMS - Interface)*   
- [ ] *api.storyblok.com (CMS - Content Delivery API)*
- [ ] *mapi.storyblok.com (CMS - Management API)*
- [ ] *capi.storyblok.com (Commerce - API)*     
- [ ] *Commerce - Interface*   
- [x] *Other:* 

**Current behavior:**
When using the code from [Webhook Secrets](https://www.storyblok.com/docs/guide/in-depth/webhooks#javascript), if your story has an & symbol in the name, it will fail authentication.

**Expected behavior:**
Users should be able to put whatever they want as their story names.

**Steps to reproduce:**
1. Setup Webhook Secrets, per the guide.
1. Create a story with an (&) symbol in the name
1. It fails authentication with the code given in the post.

**CODE FROM THE POST**
```
const crypto = require('crypto');
// signature from header 'webhook-signature'
const signature = '64c6153b5bff67......7d01f244bb7ad253e563'
// body of webhook request
const body = {
    "text": "The user ... published the Story ...",
    "action": "published",
    "space_id": 123,
    "story_id": 123
};
 
try {
  verifySignature(body, signature);
} catch(err) {
  console.log('Failure: ' + err.message);
}
 
function verifySignature(body, signature) {
  // webhookSecret configured in Storyblok settings
  const webhookSecret = process.env.STORYBLOK_WEBHOOK_SECRET; // 🙈 secrets shouldn't be in your code
  let bodyHmac = crypto.createHmac('sha1', webhookSecret)
    .update(JSON.stringify(body))
    .digest('hex');
 
  if (bodyHmac !== signature) {
    throw new Error('Signature mismatch!');
  }
}
```

**Other information:**

**Comment by bsnelder** [comment]
Hi, running into the same issue. Any solution?

**Comment by N3uRaL4Ca5t** [comment]
Buffers. Digital buffers work, working, worked. Running into which issue? The issue of mid-sentence disruptions?

**Comment by bsnelder** [comment]
Running into the issue mentioned by @hecktarzuli. Storyblok sends a different `webhook-signature` then what the example code generates for the same payload + secret. 

It works for most usecases, but when you have a `&` in the page name Storyblok sends a different signature, causing the security check to fail.

**Comment by roakoth** [comment]
Hello @hecktarzuli @bsnelder 
You need to use the raw body sent from the request or the unicode chars like the & will be decoded. 
If you are using Vercel, then you can check out this guide here:- https://vercel.com/guides/how-do-i-get-the-raw-body-of-a-serverless-function
If you are not using Vercel, you need to use the raw body.
I believe this will resolve the issue.

**Comment by hecktarzuli** [comment]
> 

Interesting, I'll check it out.  
https://flaviocopes.com/express-get-raw-body/

**Comment by bsnelder** [comment]
Yes as mentioned in the support chat using the raw body fixed this issue for me. (using pipedream). Thanks!

**Comment by alvarosabu** [comment]
Hi @hecktarzuli, we are closing this issue because this repository was used in the early days for support-related questions, but we've since transitioned to more effective channels for helping our users.

Please, refer to the following resources:

Storyblok Github Org: https://github.com/storyblok
Storyblok Docs: https://www.storyblok.com/docs
Official Support: https://support.storyblok.com
Join our Discord: https://www.storyblok.com/join-discord

Thanks for the understanding 🙏🏻
